mdoc-docs/src/main/scala/mdoc/docs/EvilplotModifier.scala

```diff
@@ -3,7 +3,10 @@ package mdoc.docs
 import com.cibo.evilplot.geometry.Drawable
 import java.nio.file.Files
 import java.nio.file.Paths
+
 import mdoc._
+import mdoc.internal.cli.{Exit, Settings}
```

Comment:
Settings can’t be exposed in the public api since it’s internal. We could add a public interface on the class if that helps

Reply:
@olafurpg Apologies for closing this but the CI/CD failed. In particular one tests assumes that the files are processed in a given order. Need to correct this.

Settings are not required so I will remove it.

Reply:
You can add `MainSettings` as an argument

https://github.com/scalameta/mdoc/blob/cf4aa2872e44a1abef1badbfe0544bf39d1a554c/mdoc/src/main/scala/mdoc/MainSettings.scala

We can add public methods down the road to access fields on the private `Settings`

Reply:
Alternatively, we can add a `abstract class Settings` in the public `mdoc` API and pass that in even if it has no public members for now. It's desirable to pass in one parameter just so we can pass in more information in the future without breaking the public API

Reply:
I don't understand the use of `MainSettings`. `Settings` does not extend this class. Note that I call `onStart` method in `MainOps`. Can you explain?

Reply:
Settings is a case class which gets auto generated methods like “copy” that break binary compatibility on removal/addition of fields. This makes case classes essentially unsuitable for public APIs unless you’re certain you will never need to add/remove a field. 

MainSettings is just a thin wrapper around the case class that has a stable api that we can evolve in a binary compatible way.

Reply:
Thanks for the explanation. `MainSettings`is now being used. I have tried to make as few changes as possible. They are basically limited to [`MainOps`](https://github.com/hmf/mdoc/blob/post-life-cycle/mdoc/src/main/scala/mdoc/internal/cli/MainOps.scala#L213). Not too happy with it because I would have liked to make the calls within the `MainOps.run`method. However this would entail changing the parameters of the constructor or `run` (`Context` is used in many places and my not be the best change). 

Please see #245